src/main/kotlin/app/extractors/Heuristics.kt

```diff
@@ -0,0 +1,316 @@
+// Copyright 2018 Sourcerer Inc. All Rights Reserved.
+// Author: Alexander Surkov (alex@sourcerer.io)
+
+package app.extractors
+
+val ActionScriptRegex = Regex(
+    "^\\s*(package\\s+[a-z0-9_\\.]+|import\\s+[a-zA-Z0-9_\\.]+;|class\\s+[A-Za-z0-9_]+\\s+extends\\s+[A-Za-z0-9_]+)"
+)
+val CSharpRegex = Regex("^\\s*namespace\\s*[\\w\\.]+\\s*\\{")
+var DRegex = Regex(
+    "^module\\s+[\\w.]*\\s*;|import\\s+[\\w\\s,.:]*;|\\w+\\s+\\w+\\s*\\(.*\\)(?:\\(.*\\))?\\s*\\{[^}]*\\}|unittest\\s*(?:\\(.*\\))?\\s*\\{[^}]*\\}"
+)
+var DTraceRegex = Regex(
+    "^(\\w+:\\w*:\\w*:\\w*|BEGIN|END|provider\\s+|(tick|profile)-\\w+\\s+\\{[^}]*\\}|#pragma\\s+D\\s+(option|attributes|depends_on)\\s|#pragma\\s+ident\\s)"
+)
+var FSharpRegex = Regex("^\\s*(#light|import|let|module|namespace|open|type)")
+var ForthRegex = Regex(" /^: /")
+var FortranRegex = Regex(
+    "^([c*][^abd-z]|      (subroutine|program|end|data)\\s|\\s*!)", RegexOption.IGNORE_CASE
+)
+var GLSLRegex = Regex("^\\s*(#version|precision|uniform|varying|vec[234])")
+var LispRegex = Regex("^\\s*\\((defun|in-package|defpackage) ", RegexOption.IGNORE_CASE)
+var MakefileRegex = Regex(
+    "([\\/\\\\].*:\\s+.*\\s\\\\$|: \\\\$|^ : |^[\\w\\s\\/\\\\.]+\\w+\\.\\w+\\s*:\\s+[\\w\\s\\/\\\\.]+\\w+\\.\\w+)"
+)
+var NewLispRegex = Regex("^\\s*\\(define ")
+val ObjectiveCRegex = Regex(
+    "^\\s*(@(interface|class|protocol|property|end|synchronised|selector|implementation)\\b|#import\\s+.+\\.h[\">])"
+)
+var Perl5Regex = Regex("\\buse\\s+(?:strict\\b|v?5\\.)")
+var Perl6Regex = Regex("^\\s*(?:use\\s+v6\\b|\\bmodule\\b|\\b(?:my\\s+)?class\\b)")
+var PythonRegex = Regex("(^(import|from|class|def)\\s)", RegexOption.MULTILINE)
+var RustRegex = Regex("^(use |fn |mod |pub |macro_rules|impl|#!?\\[)")
+var RenderScriptRegex = Regex("#include|#pragma\\s+(rs|version)|__attribute__")
+var ScalaRegex = Regex(
+    "^\\s*import (scala|java)\\./.match(data) || /^\\s*val\\s+\\w+\\s*=/.match(data) || /^\\s*class\\b"
+)
+var XMLPropsRegex = Regex("^(\\s*)(<Project|<Import|<Property|<?xml|xmlns)", RegexOption.IGNORE_CASE)
+
+val Heuristics = mapOf<String, (List<String>) -> ExtractorInterface?>(
+    "4TH" to { _ -> CommonExtractor("forth") },
```

Comment:
I think visually better to expand inlined blocks. All extractors will be aligned in the same column.